extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/DependenciesProcessor.java

```diff
@@ -0,0 +1,430 @@
+package io.quarkus.devui.deployment.menu;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+
+import io.quarkus.bootstrap.model.ApplicationModel;
+import io.quarkus.deployment.IsDevelopment;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.pkg.builditem.CurateOutcomeBuildItem;
+import io.quarkus.devui.deployment.InternalPageBuildItem;
+import io.quarkus.devui.spi.buildtime.BuildTimeActionBuildItem;
+import io.quarkus.devui.spi.page.Page;
+import io.quarkus.maven.dependency.ArtifactCoords;
+import io.quarkus.maven.dependency.ArtifactKey;
+import io.quarkus.maven.dependency.ResolvedDependency;
+
+public class DependenciesProcessor {
+
+    private static final String NAMESPACE = "devui-dependencies";
+
+    @BuildStep(onlyIf = IsDevelopment.class)
+    void createAppDeps(BuildProducer<InternalPageBuildItem> menuProducer,
+            CurateOutcomeBuildItem curateOutcomeBuildItem) {
+
+        // Menu
+
+        if (isEnabled()) {
+
+            InternalPageBuildItem page = new InternalPageBuildItem("Dependencies", 70);
+
+            page.addPage(Page.webComponentPageBuilder()
+                    .namespace(NAMESPACE)
+                    .icon("font-awesome-solid:diagram-project")
+                    .title("Application Dependencies")
+                    .componentLink("qwc-dependencies.js"));
+
+            Root root = new Root();
+            root.rootId = curateOutcomeBuildItem.getApplicationModel().getAppArtifact().toCompactCoords();
+            Set<String> allGavs = new TreeSet<>();
+            buildTree(curateOutcomeBuildItem.getApplicationModel(), root, Optional.of(allGavs), Optional.empty());
```

Comment:
My understanding is that we will always execute this in dev mode once we switch to the new incubating model.

Question: how slow is it?

Reply:
Good point. It's not really slow but would be good to avoid it. @phillip-kruger could we avoid it until we get a click on "Dependencies"?

Reply:
We can, we then need to move it from build time, where we do this at the moment, to runtime. Let me know if you want to change that and we can do it that way